### ISSUE TITLE
Nameof fix doesn't work well with Controllers

### ISSUE THREAD

**Issue by mirhagk** [code-cracker/code-cracker#595]
Let's say you have an MVC project with a controller named `ShipmentController`. If you want to create a url to the `ShipmentController` you can use something like the following:

    var url = new UrlHelper(Request.RequestContext).Action("Index/"+data.shipmentid, "Shipment");

For my case I happen to have a model called `Shipment` which seems like a normal likely scenario (a CRUD controller, and perhaps why MVC names controllers with a controller suffix). It's totally possible to do `nameof(Shipment)` and in fact code cracker suggests this, but that wouldn't help in the situation that the ShipmentController is renamed, which is exactly what the nameof operator is supposed to help with.

It could be argued that the correct code to use here would be something like:

    nameof(ShipmentController).Replace("Controller","")

Which does correctly tie this to the ShipmentController name but this doesn't seem like a good enough fix for code cracker to suggest (more convoluted, what happens to ControllerController, now suggests using nameof(Controller) which seems silly etc). I'm not sure what the correct solution would be here, but I do know that tying the variable to the wrong type doesn't seem like the correct thing to do.

**Comment by carloscds** [comment]
@mirhagk The logic under this analyzer is check if string ("Shipment in your case) is a object in code and after this suggest to use nameof(). About change controller replace "Controller" word is a different context out of scope for this analyzer. IMHO I think that nameof analyzers could be ignore suggestions in controllers for some actions.

**Comment by giggio** [comment]
@mirhagk This would be really specific. We could argue that `nameof(ShipmentController).Replace(nameof(Controller),"")` would tie things correctly, but that is really ugly. I am not sure we should do this.

**Comment by mirhagk** [comment]
@giggio Yeah I agree. The right thing here may just be not to suggest anything? Using name of(Shipment) is incorrect for sure, and there really isn't a clean solution as of yet
